operator/v1/types_console.go

```diff
@@ -143,8 +143,140 @@ type Capability struct {
 	Visibility CapabilityVisibility `json:"visibility"`
 }
 
+// ThemeMode is the value of the logo theme mode that determines the theme mode in the console UI.
+// +kubebuilder:validation:Enum="Dark";"Light"
+// +enum
+type ThemeMode string
+
+// ThemeMode values
+const (
+	// ThemeModeDark represents the dark mode for a console theme.
+	ThemeModeDark ThemeMode = "Dark"
+
+	// ThemeModeLight represents the light mode for a console theme.
+	ThemeModeLight ThemeMode = "Light"
+)
+
+// LogoType is the value of the logo type that determines if the logo is for the masthead or the favicon in the console UI.
+// The masthead logo is displayed in the masthead and about modal of the console UI.
+// +kubebuilder:validation:Enum="Masthead";"Favicon"
+// +enum
+type LogoType string
+
+const (
+	// Masthead represents the logo in the masthead.
+	LogoTypeMasthead LogoType = "Masthead"
+
+	// Favicon represents the favicon logo.
+	LogoTypeFavicon LogoType = "Favicon"
+)
+
+// SourceType defines the source type of the file reference.
+// +kubebuilder:validation:Enum="ConfigMap"
+// +enum
+type SourceType string
+
+const (
+	// SourceTypeConfigMap represents a ConfigMap source.
+	SourceTypeConfigMap SourceType = "ConfigMap"
+)
+
+// ConfigMapFileReference references a specific file within a ConfigMap.
+type ConfigMapFileReference struct {
+	// name is the name of the ConfigMap.
+	// name is a required field.
+	// Must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character.
+	// Must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+
+	// key is the logo key inside the referenced ConfigMap.
+	// Must consist only of alphanumeric characters, dashes (-), underscores (_), and periods (.).
+	// Must be at most 253 characters in length.
+	// Must end in a valid file extension.
+	// A valid file extension must consist of a period followed by 2 to 5 alpha characters.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:XValidation:rule="self.matches('.*\\\\.[a-zA-Z]{2,5}$')",message="The ConfigMap key must end with a valid file extension (2 to 5 letters)."
+	// +required
+	Key string `json:"key"`
```

Comment:
Configmap keys are validated using https://github.com/kubernetes/kubernetes/blob/381ccf0f4c723f7164d4363d2bcd76f4e34e5441/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L435-L451

Can we add another pattern validation that checks the regex here and shows that the key contains the correct character set

Reply:
> Configmap keys are validated using https://github.com/kubernetes/kubernetes/blob/381ccf0f4c723f7164d4363d2bcd76f4e34e5441/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L435-L451
> 
> Can we add another pattern validation that checks the regex here and shows that the key contains the correct character set

You mean something like this ` // +kubebuilder:validation:Pattern=`^[a-zA-Z0-9\.\-_]+$`. ` Is that right?

Reply:
Yep, exactly that